src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public static class PageDirectiveFeature
+    {
+        public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
+        {
+            const string PageDirective = "@page";
```

Comment:
Does this need to be at the very beginning of a razor page? What if my page has a whitespace character at the beginning? like `____@page` (replace _ for whitespaces)